src/http2_connection.erl

```diff
@@ -69,14 +73,15 @@
           end_headers = false       :: boolean()
 }).
 
-
 -record(stream, {
-          id :: stream_id(),
-          pid :: pid(),
-          send_window_size :: non_neg_integer(),
-          recv_window_size :: non_neg_integer(),
-          queued_data :: undefined | done | binary(),
-          body_complete = false :: boolean()
+          id                    :: stream_id(),
+          pid                   :: undefined | pid(),
+          send_window_size      :: non_neg_integer(),
+          recv_window_size      :: non_neg_integer(),
+          queued_data           :: undefined | done | binary(),
+          body_complete = false :: boolean(),
+          response_headers      :: undefined | hpack:headers(),
```

Comment:
I... do not like this. Why does the connection need to know about the headers and body in the stream?